File: daily/数组/数组easy/1. 两数之和.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
//无脑暴力 时间复杂度O(n^2)
var twoSum = function(nums, target) {
    for(var i=0;i<nums.length-1;i++){
        for(var j=i+1;j<nums.length;j++){
            if(nums[i]+nums[j]==target){
                return [i,j];
            }
        }
    }
    return [0,0];
};

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
//map 时间复杂度O(n)
var twoSum = function(nums, target) {
    var m=new Map();
    for(var i=0;i<nums.length;i++){
        if(m[target-nums[i]]!=undefined){
            return [m[target-nums[i]],i];
        }
        m[nums[i]]=i;
    }
    return [];
};

File: daily/C++版本/1. 两数之和.cpp
//ÎÞÄÔ±©Á¦·¨ 
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> box;
        for(int i=0;i<nums.size()-1;i++){
            for(int j=i+1;j<nums.size();j++){
                if(nums[i]+nums[j]==target){
                    box.push_back(i);
                    box.push_back(j);
                    return box;
                }
            }
        }
        return box;
    }
};


//¹þÏ£map
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> m;
        for(int i=0;i<nums.size();i++){
            if(m.find(target-nums[i])!=m.end()){
                return {m[target-nums[i]],i};
            }
            m[nums[i]]=i;
        }
        return {};
    }
}; 

